src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx

```diff
@@ -0,0 +1,126 @@
+  const formMethods = useForm<ExportFormData>({
+    mode: 'all',
+    defaultValues: {
+      [ExportFieldNames.password]: '',
+    },
+  })
+  const { register, formState, handleSubmit, setValue, watch, reset } = formMethods
+
+  const exportedKey = watch(ExportFieldNames.pk)
+
+  const onSubmit = async (data: ExportFormData) => {
+    try {
+      setError(undefined)
+      const pk = await exportPk(data[ExportFieldNames.password])
+      setValue(ExportFieldNames.pk, pk)
+    } catch (err) {
+      logError(ErrorCodes._305, err)
+      setError('Error exporting account. Your entered password might be invalid.')
+    }
+  }
+
+  const handleClose = () => {
+    setError(undefined)
+    reset()
+    onClose()
+  }
+  return (
+    <ModalDialog open={open} onClose={handleClose}>
+      <DialogTitle>
+        <Typography variant="h6" fontWeight={700}>
+          Export your account
+        </Typography>
+      </DialogTitle>
+      <IconButton className={css.close} aria-label="close" onClick={handleClose} size="small">
+        <Close fontSize="large" />
+      </IconButton>
+
+      <DialogContent>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <Box display="flex" flexDirection="column" gap={2} alignItems="flex-start" sx={{ width: '100%' }}>
+            <Typography>For security reasons you have to enter your password to reveal your account key.</Typography>
+
+            {exportedKey ? (
+              <Box display="flex" flexDirection="row" alignItems="center" gap={1} width="100%">
+                <TextField
+                  fullWidth
+                  multiline={showPassword}
+                  maxRows={3}
+                  label="Private key"
+                  type="password"
+                  InputProps={{
+                    readOnly: true,
+                    endAdornment: (
+                      <>
+                        <IconButton size="small" onClick={() => setShowPassword((prev) => !prev)}>
+                          {showPassword ? <VisibilityOff fontSize="small" /> : <Visibility fontSize="small" />}
+                        </IconButton>
+                        <CopyButton text={exportedKey} />
+                      </>
+                    ),
+                  }}
+                  {...register(ExportFieldNames.pk)}
+                />
+              </Box>
+            ) : (
+              <>
+                <TextField
+                  placeholder="Password"
+                  label="Password"
+                  type="password"
+                  fullWidth
+                  error={!!formState.errors[ExportFieldNames.password]}
+                  helperText={formState.errors[ExportFieldNames.password]?.message}
+                  {...register(ExportFieldNames.password, {
+                    required: true,
+                  })}
+                />
+              </>
+            )}
+            {error && <ErrorMessage className={css.modalError}>{error}</ErrorMessage>}
```

Comment:
I think the `ErrorMessage` component adds vertical margin by default which makes the UI jump a bit more. I would suggest removing it via the css class.